### ISSUE TITLE
Quesions about mean voxel density

### ISSUE THREAD

**Issue by ideasplus** [dschinagl/occam#3]
Hi, Thanks for your amazing work!

Can you please give the codes or more details on how to calculate the mean voxel density defined in  the paper? I want to adapt the method to other benchmark datasets such as waymo and nuscenes, but I don't kown how to set the following proper parameters.

https://github.com/dschinagl/occam/blob/9a3c0900f3d6172a8622cdce10a83e769ad3acbb/cfgs/occam_configs/kitti_pointpillar.yaml#L12-L15

**Comment by dschinagl** [comment]
Thank you, I appreciate your interest in our method.

To estimate the voxel density for a dataset, we used a few hundred samples from the training set. We voxelized them and determined for non-empty voxels at certain distances, e.g. every 0.5 meter, how many neighboring voxels within a radius of `1m` are non-empty. We then fitted a polynomial of the form `1 / (a*distance^2 + b*distance + c)` into these data points, thus obtaining the above parameters for KITTI for instance, see Figure 3.

The sampling probability, i.e. the probability with which a voxel is not masked, is then computed as `lambda / (a*distance^2 + b*distance + c)`. The parameter `lambda` can then be used to set the mean similarity score, i.e. how strongly the detector is challenged, see Figure 8.
